files/image_config/hostcfgd/hostcfgd

```diff
@@ -0,0 +1,339 @@
+        else:
+            tcp_port = TACPLUS_SERVER_TCP_PORT_DEFAULT
+            pri = TACPLUS_SERVER_PRI_DEFAULT
+            auth_type = None
+            timeout = None
+            passkey = None
+
+            if 'tcp_port' in data:
+                tcp_port = data['tcp_port']
+            if 'priority' in data:
+                pri = data['priority']
+            if 'auth_type' in data:
+                auth_type = data['auth_type']
+            if 'passkey' in data:
+                passkey = data['passkey']
+            if 'timeout' in data:
+                timeout = data['timeout']
+
+            found = False
+            for server in self.tacplus_servers:
+                if key == server.address:
+                    server.tcp_port = tcp_port
+                    server.auth_type = auth_type
+                    server.timeout = timeout
+                    server.passkey = passkey
+                    server.priority = pri
+                    found = True
+            if not found:
+                server = TacplusServer(key, tcp_port, passkey, timeout, auth_type, pri)
+                self.tacplus_servers.append(server)
+
+        self.tacplus_servers = sorted(self.tacplus_servers,
+                                      key=lambda server: server.priority,
+                                      reverse=True)
+        if modify_conf:
+            self.modify_pam_auth_file()
+            self.modify_nss_conf()
+
+    def get_pam_auth(self, pam_module):
+        if pam_module == AAA_LOCAL:
+            pam_auth = PAM_LOCAL_MODULE
+            if self.debug:
+                pam_auth += " debug"
+            pam_auth += " nullok try_first_pass\n"
+            return pam_auth
+        else:
+            return pam_module.get_pam_conf(self.tacplus_global, self.debug)
+
+    def modify_pam_auth_file(self):
+        if self.auth_failthrough:
+            pam_control = "[success=done new_authtok_reqd=done default=ignore]"
+        else:
+            pam_control = "[success=done new_authtok_reqd=done default=ignore auth_err=die]"
+
+        pam_modules = []
+        auth_file_body = ""
+
+        # Set local and tacacs+ pam priority
+        # ['local'] or ['local','tacacs+'] or ['tacacs+', 'local'] or ['tacacs+']
+        if self.auth_login == [AAA_LOCAL]:
+            pam_modules = [AAA_LOCAL]
+        elif self.auth_login == [AAA_LOCAL, AAA_TACPLUS]:
+            pam_modules = [AAA_LOCAL] + self.tacplus_servers
+        elif self.auth_login == [AAA_TACPLUS] or self.auth_login == [AAA_TACPLUS, AAA_LOCAL]:
+            # Don't accept only TACACS+ authentication
+            # Make sure root will always authentication on local, not TACACS+
+            pam_modules = self.tacplus_servers + [AAA_LOCAL]
+            auth_file_body += "auth\t[success=%d new_authtok_reqd=done default=ignore]\t" % (len(pam_modules)-1)
+            auth_file_body += "pam_succeed_if.so user = root debug\n"
```

Comment:
root is not allowed to login by default, there is also no debug account, only admin is enabled.

Reply:
This configuration is used to ensure that an administrator user always login via local if TACACS+ authentication hangs or other error occur. But it's not correct to use root. The 'debug' means output debug log, not means debug account.
I have removed this logic because it looks like an impossible scenario.